engine/src/main/java/org/pentaho/di/pan/TransParams.java

```diff
@@ -0,0 +1,98 @@
+/*! ******************************************************************************
+ *
+ * Pentaho Data Integration
+ *
+ * Copyright (C) 2002-2019 by Hitachi Vantara : http://www.pentaho.com
+ *
+ *******************************************************************************
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ ******************************************************************************/
+
+package org.pentaho.di.pan;
+
+import org.pentaho.di.base.Params;
+import org.pentaho.di.base.KettleConstants;
+import org.apache.commons.lang.StringUtils;
+import org.pentaho.di.core.parameters.NamedParams;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class TransParams extends Params {
+
+
+    public TransParams( String blockRepoConns, String repoName, String repoUsername, String trustRepoUser, String repoPassword, String inputDir,
+                        String inputFile, String listRepoFiles, String listRepoDirs, String exportRepo, String localFile,
+                        String localJarFile, String localInitialDir, String listRepos, String safeMode, String metrics,
+                        String listFileParams, String logLevel, String maxLogLines, String maxLogTimeout, String logFile,
+                        String oldLogFile, String version, String resultSetStepName, String resultSetCopyNumber, NamedParams namedParams ) {
+
+        super( blockRepoConns, repoName, repoUsername, trustRepoUser, repoPassword, inputDir, inputFile, listRepoFiles, listRepoDirs, exportRepo,
+                localFile, localJarFile, localInitialDir, listRepos, safeMode, metrics, listFileParams, logLevel, maxLogLines,
+                maxLogTimeout, logFile, oldLogFile, version, resultSetStepName, resultSetCopyNumber, namedParams, null );
+    }
+
+    @Override
+    public Map<String, String> asMap() {
+
+        Map<String, String> arguments = new HashMap<>();
+
+        putIfNotEmpty( arguments, "uuid", getUuid() );
```

Comment:
we can replace these `putIfNotEmpty( arguments, key, value );` lines with `arguments.putIfAbsent( key, value )` and forgo that private `putIfNotEmpty` method